boards/PDM/Src/Io/Io_Aux1Aux2Efuse.c

```diff
@@ -0,0 +1,137 @@
+#include "Io_Aux1Aux2Efuse.h"
+#include "configs/Io_EfuseConfig.h"
+
+static struct Efuse_Context *aux1_aux2_efuse;
+
+void Io_Aux1Aux2Efuse_Init(SPI_HandleTypeDef *const hspi)
+{
+    assert(hspi != NULL);
+
+    aux1_aux2_efuse = Io_Efuse_Create(
+        hspi, CSB_AUX1_AUX2_GPIO_Port, CSB_AUX1_AUX2_Pin,
+        FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin, FSB_AUX1_AUX2_GPIO_Port,
+        FSB_AUX1_AUX2_Pin, PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, PIN_AUX2_GPIO_Port,
+        PIN_AUX2_Pin);
+}
+
+void Io_Aux1Aux2Efuse_EnableAux1(void)
+{
+    Io_Efuse_EnableChannel0(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_DisableAux1(void)
+{
+    Io_Efuse_DisableChannel0(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_EnableAux2(void)
+{
+    Io_Efuse_EnableChannel1(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_DisableAux2(void)
+{
+    Io_Efuse_DisableChannel1(aux1_aux2_efuse);
+}
+
+ExitCode Io_Aux1Aux2Efuse_GetStatus(enum Efuse_Status *status)
+{
+    return Io_Efuse_ReadRegister(
+        SO_STATR_ADDR, (uint16_t *)status, aux1_aux2_efuse);
+}
+
+ExitCode Io_Aux1Aux2Efuse_GetAux1Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_ReadRegister(
+        SO_FAULTR_0_ADDR, (uint16_t *)fault, aux1_aux2_efuse);
+}
+
+ExitCode Io_Aux1Aux2Efuse_GetAux2Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_ReadRegister(
+        SO_FAULTR_1_ADDR, (uint16_t *)fault, aux1_aux2_efuse);
+}
+
+bool Io_Aux1Aux2Efuse_IsInFaultMode(void)
+{
+    return Io_Efuse_IsEfuseInFaultMode(aux1_aux2_efuse);
+}
+
+bool Io_Aux1Aux2Efuse_IsInFailSafeMode(void)
+{
+    return Io_Efuse_IsEfuseInFailSafeMode(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_DelatchFaults(void)
+{
+    Io_Efuse_DelatchFaults(aux1_aux2_efuse);
+}
+
+float Io_Aux1Aux2Efuse_GetAux1Current(void)
```

Comment:
Just giving this a 2nd look. This function can be moved to the Io_Efuse layer. 

- AUX1/2_CURRENT_SENSE_CHANNEL and the rest of the current sense channels for the remaining efuses have the same value (CSNS_FUNCTION_CURRENT_CH0/1, regardless of the efuse).
- Io_CurrentSense can be passed as a function pointer to the Efuse layer through Io_Efuse_Create
- CUR_SYNC gpio can be passed to the Efuse layer just like the previous GPIOs

Reply:
🧠 moment

Reply:
🥔 🧠 moment for me

Reply:
fixed